### PR TITLE
Add test for apm footer to not apply template if = empty string

### PR DESCRIPTION
**Pull request by rothgar**
This causes an error on nodes that don't have an apm footer set. The default value is an empty string and tests between puppet 3 and puppet 4 are different.
Explicitly testing for an empty string to know if we should create the template.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -451,10 +451,12 @@
       $apm_footer_order = '06'
     }
 
-    concat::fragment{ 'datadog apm footer':
-      target  => '/etc/dd-agent/datadog.conf',
-      content => template('datadog_agent/datadog_apm_footer.conf.erb'),
-      order   => $apm_footer_order,
+    if $apm_env != '' {
```
Comment (truthbk) [inline manifests/init.pp]: would it maybe make more sense to check for `$apm_enabled`?

Comment (rothgar) [reply manifests/init.pp]: That makes sense but if apm_enabled == true but apm_env is not set then you would still get an error.

I could add a check for both so long as we think the user will know they need to set both variables to have it configured.
